test/unit/math/opencl/device_functions/lbeta_test.cpp

```diff
@@ -0,0 +1,56 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <stan/math/opencl/kernels/device_functions/lgamma_stirling.hpp>
+#include <stan/math/opencl/kernels/device_functions/lgamma_stirling_diff.hpp>
+#include <stan/math/opencl/kernels/device_functions/lbeta.hpp>
+#include <stan/math/opencl/kernel_cl.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/expect_near_rel.hpp>
+#include <string>
+
+static const std::string test_kernel_code = STRINGIFY(__kernel void test(
+    __global double *C, __global double *B, __global double *A) {
+  const int i = get_global_id(0);
+  C[i] = lbeta(A[i], B[i]);
+});
+
+const stan::math::opencl_kernels::kernel_cl<
+    stan::math::opencl_kernels::out_buffer,
+    stan::math::opencl_kernels::in_buffer,
+    stan::math::opencl_kernels::in_buffer>
+    lbeta("test",
+          {stan::math::opencl_kernels::lgamma_stirling_device_function,
+           stan::math::opencl_kernels::lgamma_stirling_diff_device_function,
+           stan::math::opencl_kernels::lbeta_device_function,
+           test_kernel_code});
```

Comment:
I don't like how I have to include here the sources for all functions lbeta depends on. If someone has an idea how we could put this within the lbeta source or make it nicer in some other way I am open to suggestions.